backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/impl/ElasticsearchSearchQueryImpl.java

```diff
@@ -141,23 +144,32 @@ public long fetchTotalHitCount() {
 		}
 		builder.query( filteredPayload )
 				.routingKeys( routingKeys )
-				.timeout( timeoutValue, timeoutUnit, exceptionOnTimeout )
+				.timeout( timeoutManager )
 				.requestTransformer(
 						ElasticsearchSearchRequestTransformerContextImpl.createTransformerFunction( requestTransformer )
 				);
 		NonBulkableWork<Long> work = builder.build();
-		return Futures.unwrappedExceptionJoin( queryOrchestrator.submit( work ) );
+		Long result = Futures.unwrappedExceptionJoin( queryOrchestrator.submit( work ) );
+		timeoutManager.stop();
+		return result;
 	}
 
 	@Override
 	public ElasticsearchSearchScroll<H> scroll(int chunkSize) {
+		timeoutManager.start();
```

Comment:
I just noticed this... you're not supposed to use the timeout manager during the initial call to `scroll()`, since we don't send a request to Elasticsearch.

Reply:
Actually with the initial call of scroll() on Elasticsearch we got the first result page of next() you know. So, IMO makes sense to keep it.

Reply:
Done and more important this remove discover an inner bug solved by this commit https://github.com/hibernate/hibernate-search/pull/2333/commits/6e95d93879d1d99a0b53e26bfe2c1b7154bd608d.